src/main.rs

```diff
@@ -552,6 +576,38 @@ fn run() -> Result<(), failure::Error> {
         } else {
             commands::subdomain::get_subdomain(&user, &target)?;
         }
+    } else if let Some(route_matches) = matches.subcommand_matches("route") {
+        let user = settings::global_user::GlobalUser::new()?;
+        let manifest = settings::toml::Manifest::new(config_path)?;
+        let env = matches.value_of("env");
+
+        let env_zone_id = if let Some(environment) = manifest.get_environment(env)? {
+            environment.zone_id.as_ref()
+        } else {
+            None
+        };
+
+        let zone_id: Result<String, failure::Error> = if let Some(zone_id) = env_zone_id {
+            Ok(zone_id.to_string())
+        } else if let Some(zone_id) = manifest.zone_id {
+            Ok(zone_id)
+        } else {
+            failure::bail!(
+                "You must specify a zone_id in `wrangler.toml` to use `wrangler route` commands."
+            )
+        };
```

Comment:
this could honestly be a really nice Manifest method that we call here?

Reply:
it's a one-off, so i'm inclined not to shove this into Manifest, or at minimum to wait until it repeats. It already kinda bothers me that deploy target is there, and honestly i think if i go for refactors i will turn that method inside out.